framework/src/modules/token/cc_commands/cc_transfer.ts

```diff
@@ -49,104 +46,140 @@ export class CCTransferCommand extends BaseCCCommand {
+				throw error;
+			}
 		}
 
-		if (
-			!tokenSupported(this._supportedTokenIDs, params.tokenID) &&
-			ccm.fee >= ctx.ccmSize * MIN_RETURN_FEE &&
-			ccm.status === CCM_STATUS_OK
-		) {
-			await this._interopMethod.error(methodContext, ccm, CCM_STATUS_TOKEN_NOT_SUPPORTED);
+		if (status !== CCM_STATUS_OK) {
+			recipientAddress = senderAddress;
+		}
+
+		const userStore = this.stores.get(UserStore);
+		const recipientExist = await userStore.has(
+			methodContext,
+			userStore.getKey(recipientAddress, tokenID),
+		);
+
+		if (!recipientExist) {
+			if (fee < FEE_CCM_INIT_USER_STORE) {
+				throw new Error('Insufficient fee to initialize user account.');
+			}
+
+			// should replace feeAddress with trsSender address
+			await this._tokenMethod.burn(methodContext, feeAddress, tokenID, amount);
 		}
 
-		// let { recipientAddress } = params;
-		// if (ccm.status !== CCM_STATUS_OK) {
-		// 	recipientAddress = params.senderAddress;
-		// }
-
-		// const canonicalTokenID = await this._tokenMethod.getCanonicalTokenID(
-		// 	methodContext,
-		// 	params.tokenID,
-		// );
-		// const recipientExist = await this._tokenMethod.accountExists(
-		// 	methodContext,
-		// 	params.recipientAddress,
-		// );
-		// let receivedAmount = params.amount;
-		// if (!recipientExist) {
-		// 	const minBalance = this._minBalances.find(mb => mb.tokenID.equals(canonicalTokenID))?.amount;
-		// 	if (!minBalance || minBalance > params.amount) {
-		// 		if (ccm.fee >= MIN_RETURN_FEE * ctx.ccmSize && ccm.status === CCM_STATUS_OK) {
-		// 			await this._interopMethod.error(methodContext, ccm, CCM_STATUS_MIN_BALANCE_NOT_REACHED);
-		// 		}
-		// 		return;
-		// 	}
-		// 	receivedAmount -= minBalance;
-		// 	const [canonicalChainID, canonicalLocalID] = splitTokenID(canonicalTokenID);
-		// 	if (canonicalChainID.equals(CHAIN_ID_ALIAS_NATIVE)) {
-		// 		const supplyStore = this.stores.get(SupplyStore);
-		// 		const supply = await supplyStore.get(methodContext, canonicalLocalID);
-		// 		supply.totalSupply -= minBalance;
-		// 		await supplyStore.set(methodContext, canonicalLocalID, supply);
-		// 	}
-		// }
-
-		// if (tokenChainID.equals(ownChainID)) {
-		// 	const escrowStore = this.stores.get(EscrowStore);
-		// 	const escrowKey = Buffer.concat([ccm.sendingChainID, tokenLocalID]);
-		// 	const escrowData = await escrowStore.get(methodContext, escrowKey);
-
-		// 	escrowData.amount -= params.amount;
-		// 	await escrowStore.set(methodContext, escrowKey, escrowData);
-		// }
-
-		// const userStore = this.stores.get(UserStore);
-		// await userStore.addAvailableBalanceWithCreate(
-		// 	methodContext,
-		// 	recipientAddress,
-		// 	canonicalTokenID,
-		// 	receivedAmount,
-		// );
+		try {
+			if (tokenChainID.equals(ownChainID)) {
+				const escrowStore = this.stores.get(EscrowStore);
+				const escrowKey = escrowStore.getKey(sendingChainID, tokenID);
+				const escrowData = await escrowStore.get(methodContext, escrowKey);
+
+				if (escrowData.amount > amount) {
```

Comment:
Would;
```if (escrowData.amount < ) throw ... ``` be enough?

Reply:
The logic here is 

<img width="450" alt="image" src="https://user-images.githubusercontent.com/10962699/197286425-7cc2a5d2-6345-4b78-9c95-6db86bb8eb68.png">

Once the a deduction happens, I update the escrow store with the new amount. By only checking that the escrow amount is < than the amount, we don't track the escrow amount. IMHO.

Reply:
I think what @has5aan meant was if/else if is not necessary, and can be done like below?
```
if (escrowData.amount < amount) {
  throw xxx
}
escrowData.amount -= amount;
...
```

Reply:
Ahh makes sense now 🙂